File: AngularAndNetCoreAuth/Models/Payment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace AngularAndNetCoreAuth.Models
{
    public class Payment : BaseEntity
    {
        // refers to ID in srms student table

        public  string Indexno { get; set; }
        public  string Name { get; set; }
        public  string Email { get; set; }
        
        public Guid ProductId { get; set; }
        
        [Required] public decimal Amount { set; get; }
        [Required] public string Bank { get; set; }

        
        [Required] public string AcademicYear { get; set; }

        public string PaymentRemarks { get; set; }

        // actual phone no used for the payment
        [Required] public string Phone { get; set; }
        [Required] public bool Status { get; set; }
        [Required] public Guid TransactionId { get; set; }

        [Required] public string WalletType { get; set; } // eg mtn, voda, airtel, mastercard...

        // actual date the client initiated the transaction
        [DataType(DataType.DateTime)] public DateTime TransactionDate { get; set; } = DateTime.UtcNow;

        // date the bank push transaction to us
        [DataType(DataType.DateTime)] public DateTime BankDate { get; set; }
    }
}

File: New_Service/backend/src/main/java/com/ttu/pay/model/Payment.java
package com.ttu.pay.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;
@Getter
@Setter
@Accessors(chain=true)
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "payments", uniqueConstraints = {
        @UniqueConstraint(columnNames = "TransactionId")
})
public class Payment implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", nullable = false)
    private UUID id;

    @NotNull
    private String indexno;
    @NotNull
    private String name;
    @Email
    @NotNull
    private String email;
    @NotNull
    private String level;
    @NotNull
    private BigDecimal amount;
    @NotNull
    private String bank;
    @NotNull
    private String academicYear;
    @NotNull
    private String paymentRemarks;
    @NotNull
    // actual phone no used for the payment
    private String phone;
    @NotNull
    private boolean status;
    @NotNull
    @Column(unique = true)
    private String transactionId;
    @NotNull
    private Providers walletType = Providers.MTN;
    // actual date the client initiated the transaction
    @NotNull
    private LocalDateTime transactionDate;
    @NotNull
    // date the bank push transaction to us
    private LocalDateTime bankDate;
    private LocalDateTime CreatedOn;
    private LocalDateTime UpdatedOn;
    @ManyToOne
    @JoinColumn(name = "user_id")
    @NotNull
    public User user;

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }
}
